### PR TITLE
fix(scroll): fix scrolling behavior | Issue #854

### LINKED ISSUE #854: BackToTop Button Scrolling Issue
On observing, I have found that the backtotop button working is not smooth, making a bad experience for user.

https://github.com/user-attachments/assets/7e4071b3-ab6f-407c-9fc5-583725a2c92f

@anmode Please assign this to me.

### PR DESCRIPTION
**Pull request by Shariq2003**
## Related Issue

https://github.com/user-attachments/assets/c006ec96-3bd5-4ca1-80c0-fc44b72070a1

TimeStamp : 
00:00-00:09 -> Before Changes
00:09-onwards -> After Changes

Closes: #854 

## Description of Changes
 Just added the scroll speed controller that adjust scrolling speed according to Y value or height of the window.
@Sanchitbajaj02 please review
## Checklist:

- [X] My code adheres to the established style guidelines of this project.
- [X] I have conducted a self-review of my code.
- [X] I have included comments in areas that may be difficult to understand.
- [X] I have made corresponding updates to the project documentation.
- [X] My changes have not introduced any new warnings.

### CODE REVIEW

**Comment by Shariq2003** [comment]
@Sanchitbajaj02  please review

File: components/basic/MoveToTop.js [hunk 1]
```diff
@@ -19,10 +19,15 @@ const ScrollButton = () => {
   };
 
   const scrollToTop = () => {
-    window.scrollTo({
-      top: 0,
-      behavior: "smooth",
-    });
+    const scrollDuration = 500;
```
Comment (Sanchitbajaj02) [inline components/basic/MoveToTop.js]: The improvement looks fine to me but do we actually need the custom scroll implementation?

File: components/basic/MoveToTop.js [hunk 2]
```diff
@@ -19,10 +19,15 @@ const ScrollButton = () => {
...
+    const scrollInterval = setInterval(() => {
+      if (window.scrollY === 0) {
+        clearInterval(scrollInterval); 
+      } else {
+        window.scrollBy(0, scrollStep); 
+      }
+    }, 15);
   };
 
   window.addEventListener("scroll", toggleVisible);
```
Comment (Sanchitbajaj02) [inline components/basic/MoveToTop.js]: Can you optimize the functionality by removing the event listener from dom by using the useEffect? Maybe we actually don't need a custom scroll after that....

**Review by Sanchitbajaj02 — CHANGES_REQUESTED** [review]

File: components/basic/MoveToTop.js (hunk 2 above)
Comment (Shariq2003) [reply components/basic/MoveToTop.js]: Sure

**Comment by Shariq2003** [comment]
I will update by today

**Comment by Shariq2003** [comment]
@Sanchitbajaj02 please review and merge

**Review by Sanchitbajaj02 — APPROVED** [review]
LGTM

**Comment by Shariq2003** [comment]
@Sanchitbajaj02 review and please merge

**Pull request merged**